### ISSUE TITLE
apply-rke2-worker-plan on windows nodes fails

### ISSUE THREAD

**Issue by faikozgur** [rancher/rke2#2998]
**Environmental Info:**
RKE2 Version: 1.22.9

Node(s) CPU architecture, OS, and Version: 

Windows Server 2019 Datacenter Ver 1809 Build 17763:2989

Cluster Configuration:

1 server node, 2 linux and one windows agents. All are installed on Hyper-V testbed.

**Describe the bug:**

Repeatedly observe a failing job/pod named apply-rke2-worker-plan* in rancher UI events list. 
Here is the events logged for a run of the job (rke2-win-wrk1 is the hostname of the windows node):

BackOff (41)	17 mins ago	Back-off pulling image "rancher/rke2-upgrade:v1.22.9-rke2r2"
Failed (4)	26 mins ago	Error: ErrImagePull
Pulling (4)	26 mins ago	Pulling image "rancher/rke2-upgrade:v1.22.9-rke2r2"
Scheduled	27 mins ago	Successfully assigned cattle-system/apply-rke2-worker-plan-on-rke2-win-wrk1-with-1f180fd65--1-f2265 to rke2-win-wrk1

"rancher/rke2-upgrade:v1.22.9-rke2r2" image doesn't have a windows os distribution. That's why image pulling stage fails.

**Steps To Reproduce:**

- Installed RKE2: 
Install RKE2 following "Quick Start" section from documentation (https://docs.rke2.io/install/quickstart/)
Add nodes following the instructions. One node should be Windows 2019.

**Expected behavior:**

**Actual behavior:**

**Additional context / logs:**

**Comment by brandond** [comment]
I'm a bit confused, you mention installing RKE2 manually using the quickstart docs, but you also appear to be managing the cluster via Rancher. Is this an imported cluster, or did you provision it using Rancher? What version of Rancher are you using? RKE2 itself does not come with any plans, so this is probably going to be a Rancher issue.

**Comment by faikozgur** [comment]
> I'm a bit confused, you mention installing RKE2 manually using the quickstart docs, but you also appear to be managing the cluster via Rancher. Is this an imported cluster, or did you provision it using Rancher? What version of Rancher are you using? RKE2 itself does not come with any plans, so this is probably going to be a Rancher issue.

@brandond I am sorry for incomplete issue report. You are right. I have also installed Rancher 2.6 by following the steps from section "Install Rancher with Helm" of https://rancher.com/docs/rancher/v2.6/en/quick-start-guide/deployment/quickstart-manual-setup/ documents. That was the last step to reproduce the issue. 
And now I am suspecting that I have opened the issue in wrong repository. I am new of experimenting Rancher products family. The "..rke2.."  part of the job name caused the confussion. 
Should I open the issue in rancher repo?

**Comment by phillipsj** [comment]
@faikozgur we can move this for you.  I will dig into this and see what I can find. I will probably have some follow questions.  I use hyper-v daily so I should get close to what is happening.

**Comment by phillipsj** [comment]
@faikozgur what is your exact Rancher version?  Did you install Rancher then upgrade it?  Did you try to upgrade the cluster?

**Comment by faikozgur** [comment]
@phillipsj 
I have just followed the steps from the "Install Rancher with Helm" section of the documents.  I have skipped the previous k8s installation steps.  Also I haven't use a full domain name for "--set hostname=<IP_OF_LINUX_NODE>.sslip.io " setting. Just setted it as :  "--set hostname=rke2-master "
Here is the version page of my Rancher setup:
![rv](https://user-images.githubusercontent.com/36256916/170795164-54cf414c-2c99-4af8-8665-a5e06b5f93f7.png)
